webofneeds/won-owner-webapp/src/main/webapp/config/details/payment.js

```diff
@@ -0,0 +1,154 @@
+import { generateIdString } from "../../app/utils.js";
+//import Immutable from "immutable";
+import won from "../../app/won-es6.js";
+import { get } from "../../app/utils.js";
+
+export const paypalPayment = {
+  identifier: "paypalPayment",
+  label: "PayPal Payment",
+  icon: "#ico36_detail_price", //TODO: ADAPT
+  amountLabel: "Price:",
+  amountPlaceholder: "Enter Amount...",
+  receiverLabel: "Recipient:",
+  receiverPlaceholder: "PayPal Account ID (Email)...",
+  secretLabel: "Secret:",
+  secretPlaceholder: "Enter Secret...",
+  customerLabel: "Who pays?",
+  component: "won-paypal-payment-picker",
+  viewerComponent: "won-paypal-payment-viewer",
+  currency: [
+    { value: "EUR", label: "€", default: true },
+    { value: "USD", label: "$" },
+    { value: "GBP", label: "£" },
+    //TODO: ADAPT
+  ],
+  messageEnabled: true,
+  parseToRDF: function({ value, identifier, contentUri }) {
+    //TODO: IMPL
+    if (
+      !value ||
+      !value.amount ||
+      !value.currency ||
+      !value.secret ||
+      !value.receiver ||
+      !value.customerUri
+    ) {
+      return { "s:invoice": undefined };
+    }
+
+    const idString =
+      contentUri && identifier
+        ? contentUri + "/" + identifier + "/" + generateIdString(10)
+        : undefined;
+    return {
+      "s:invoice": {
+        "@id": idString,
+        "@type": "s:Invoice",
+        "s:paymentMethod": {
+          "@type": "s:PaymentMethod",
+          "@id": "gr:PayPal",
+        },
+        "s:paymentStatus": { "@id": "s:PaymentDue" },
+        "s:totalPaymentDue": {
+          "@id": idString ? idString + "#totalPaymentDue" : undefined,
+          "@type": "s:CompoundPriceSpecification",
+          "s:price": [{ "@value": value.amount, "@type": "xsd:float" }],
+          "s:priceCurrency": value.currency,
+        },
+        "s:paymentMethodId": value.secret, //TODO not sure if this would be the correct predicate for our Secret
```

Comment:
[s:identifier](https://schema.org/identifier) might be a better fit than s:paymentMethodId here.